### ISSUE TITLE
Adding multiple similar connections doesn't work correctly

### ISSUE THREAD

**Issue by kburtram** [microsoft/vscode-mssql#18001]
Steps to Reproduce:

1. Create a saved connection to a database on a server using SqlLogin
2. Create a saved connection to the server using Entra ID

Expected: Two distinct connections, one to DB and one to Server with different auth types.  Clearly distinguishable in connection display name.  Note: I'm not sure the repro needs to be this specific, probably any slightly ambiguous connections will have similar problems.

Actual: Two connections have same title, one of them clearly wrong.  The contents in the below screenshot show both a DB and Server type connection (based on the subnodes), but the display names are identical.  The second one being incorrect.

![image](https://github.com/user-attachments/assets/15758074-3a41-4d18-b69e-d0807c8a9ecf)

**Comment by kburtram** [comment]
This is similar repro when adding a connection from the MRU.  I had two entries added to settings JSON for same the connection, one with only minimal details.  They both show in Connections list but can't tell them apart.  

Also, I don't think those generated profileNames should be put into the settings file since they are supposed to be auto-generated based on connection metadata.  For example, if the metadata changes will the saved name be overridden with the new data & how can we tell the difference between the auto-gen name and user provided "friendly name"?

![image](https://github.com/user-attachments/assets/48063e89-e314-439b-b72c-0865d54476f3)

**Comment by Benjin** [comment]
> I don't think those generated profileNames should be put into the settings file since they are supposed to be auto-generated based on connection metadata

Agreed; fixed as part of https://github.com/microsoft/vscode-mssql/pull/18030

**Comment by Benjin** [comment]
This should be mostly resolved with https://github.com/microsoft/vscode-mssql/pull/18684, though I still need to close some corner cases, like ensuring the displayname is differentiated.
